### PR TITLE
os/fs/procfs: Add up current and peak heap in all heaps

### PR DESCRIPTION
**Pull request by zhouxinhe**
In current logic, it only get the usage from one heap.
It's not exact, when user configure multiple heaps and
the task/thread may allocate memory from different heaps.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a4035cc4ea1908e400c925465f846a4e05c67be4] -

**Comment by seinfra** [comment]
Target : [a4035cc4ea1908e400c925465f846a4e05c67be4] -

File: os/fs/procfs/fs_procfsproc.c [hunk 1]
```diff
@@ -381,24 +382,24 @@ static ssize_t proc_entry_stat(FAR struct proc_file_s *procfile, FAR struct tcb_
...
-		/* Idle task normally uses heap with index 0. */
-		heap = mm_get_heap_with_index(0);
-	} else {
-		heap = mm_get_heap(tcb->stack_alloc_ptr);
-	}
-	if (heap == NULL) {
-		return -1;
-	}
-
+	curr_heap = 0;
```
Comment (SRCN-LiQin) [inline os/fs/procfs/fs_procfsproc.c]: this 0 initialization could refine the logic of += using in for loop, but it's changing the default -1 value when there is no alloc_size for current tcb. how about initialize it to -1 as before, and check it after for loop to revise the -1

Comment (zhouxinhe) [reply os/fs/procfs/fs_procfsproc.c]: If CONFIG_DEBUG_MM_HEAPINFO is not defined, the default value is -1 (feature not support). But if it's defined, -1 is an illegal value (the minimal value is 0).

**Comment by seinfra** [comment]
Target : [0d0714d007103d4de03696b6a940bc92c0f6715e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0d0714d007103d4de03696b6a940bc92c0f6715e] - Code Rule Check (C++) OK.

**Comment by zhouxinhe** [comment]
@jeongchanKim Could you please help to review?

**Pull request closed without merging**